src/process_utility.c

```diff
@@ -1747,6 +1655,80 @@ process_ddl_command_end(CollectedCommand *cmd)
 	}
 }
 
+static void
+process_drop_constraint_on_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
+{
+	char	   *hypertable_constraint_name = arg;
+	Chunk	   *chunk = chunk_get_by_relid(chunk_relid, ht->space->num_dimensions, true);
+
+	/* drop both metadata and table; sql_drop won't be called recursively */
+	chunk_constraint_delete_by_hypertable_constraint_name(chunk->fd.id, hypertable_constraint_name, true, true);
+}
+
+static void
+process_drop_table_constraint(EventTriggerDroppedObject *obj)
+{
+	/* addrnames: schema, table, constraint_name */
+
+	/* do not use relids because underlying table could be gone */
+	Hypertable *ht = hypertable_get_by_name(linitial(obj->addrnames), lsecond(obj->addrnames));
+
+	if (ht != NULL)
+	{
+		CatalogSecurityContext sec_ctx;
+
+		catalog_become_owner(catalog_get(), &sec_ctx);
+
+		/* Recurse to each chunk and drop the corresponding constraint */
+		foreach_chunk(ht, process_drop_constraint_on_chunk, lthird(obj->addrnames));
+
+		catalog_restore_user(&sec_ctx);
+	}
+	else
+	{
+		Chunk	   *chunk = chunk_get_by_name(linitial(obj->addrnames), lsecond(obj->addrnames), 0, false);
+
+		if (NULL != chunk)
+		{
+			chunk_constraint_delete_by_constraint_name(chunk->fd.id, lthird(obj->addrnames), true, false);
+		}
+	}
+}
+
+static void
+drop_index(EventTriggerDroppedObject *obj)
```

Comment:
`process_drop_index()`? Also, this function seems to be a somewhat unnecessary indirection.